docker/docker_client.go

```diff
@@ -81,10 +81,20 @@ type bearerToken struct {
 // dockerClient is configuration for dealing with a single Docker registry.
 type dockerClient struct {
 	// The following members are set by newDockerClient and do not change afterwards.
-	sys                   *types.SystemContext
-	registry              string
-	client                *http.Client
-	insecureSkipTLSVerify bool
+	sys                  *types.SystemContext
+	registry             string
+	sysContextSkipVerify *bool
+
+	// The httpTransport is setup during the creation of the dockerClient and
+	// will be updated by setting the TLS config (InsecureSkipVerify) before
+	// doing any request. Any HTTP request the dockerClient does will be done
+	// by this transport.
+	httpTransport *http.Transport
+
+	// The endpointSkipVerify will be used as a fallback if
+	// sysContextSkipVerify == nil.  It will be initially set by
+	// newDockerClient but can be overwritten afterwards.
+	endpointSkipVerify bool
```

Comment:
I don't think this is necessary. The dockerClient already has an `insecureSkipTLSVerify` and the `systemContext` has priority. Instead of deferring the decision, the logic of `skipVerify()` should be done when creating a client.